lib/AgrirouterClient/src/Registration.cpp

```diff
@@ -10,17 +11,18 @@
 #include <iostream>
 #include <vector>
 
-Registration::Registration(ConnectionProvider *connectionProvider, Settings *settings)
+Registration::Registration(ConnectionProvider *connectionProvider, Settings *settings, void *member)
 {
     m_settings = settings;
     m_connectionProvider = connectionProvider;
+    m_member = member;
 }
 
 Registration::~Registration() {}
 
 void Registration::registerToAgrirouterWithRegCode(std::string& registrationCode, AgrirouterSettings& agrirouterSettings)
 {
-    this->m_registrationCode = registrationCode;
+    m_registrationCode = registrationCode;
```

Comment:
registerToAgrirouterWithRegCode() can either be merged into sendOnboard() or sendOnboard() should be private.